app/models/tag.rb

```diff
@@ -23,6 +23,69 @@ class Tag < ActiveRecord::Base
   # the order is important, and it is the order in which they appear in the tag wrangling interface
   USER_DEFINED = ['Fandom', 'Character', 'Relationship', 'Freeform']
 
+  def check_if_large_tag(length_of_time_in_cache,how_long_it_takes_to_compute_size)
+    if length_of_time_in_cache < 40 && how_long_it_takes_to_compute_size < (ArchiveConfig.TAGGINGS_COUNT_DEFAULT_FOR_LARGE_TAGS || 4)
+      if self.large_tag
+        self.large_tag = false
+        self.save
+      end
+      return
+    end
+    return if self.large_tag
+    self.large_tag = true
+    self.save
+  end
+
+  def taggings_count_expiry(count)
+    expiry_time = count/(ArchiveConfig.TAGGINGS_COUNT_CACHE_DIVISOR || 2000 )
+    [[expiry_time, (ArchiveConfig.TAGGINGS_COUNT_MIN_TIME||3)].max, (ArchiveConfig.TAGGINGS_COUNT_MAX_TIME||60)].min
+  end
+
+  def taggings_count_cache_key
+    "/v1/taggings_count/#{self.id}"
+  end
+
+  def taggings_count=(value)
+    expiry_time = taggings_count_expiry(value)
+    Rails.cache.write(taggings_count_cache_key, value, race_condition_ttl: 10, expires_in: expiry_time.minutes) if (value > (ArchiveConfig.TAGGINGS_COUNT_MIN_CACHE_COUNT || 1000))
+    check_if_large_tag(expiry_time,0)
+    if self.taggings_count_cache != value
+      self.taggings_count_cache = value
+      self.save
+    end
+  end
+
+  def taggings_count
+    cache_read = Rails.cache.read(self.taggings_count_cache_key)
+    return cache_read unless cache_read.nil?
+    time_start = Time.now.to_i
+    real_value = self.taggings.length
+    time_end = Time.now.to_i
+    if (real_value > (ArchiveConfig.TAGGINGS_COUNT_MIN_CACHE_COUNT || 1000)) || (time_end - time_start > (ArchiveConfig.TAGGINGS_COUNT_MAX_ALLOWED_TIME || 6))
+      self.taggings_count = real_value
+      check_if_large_tag(taggings_count_expiry(real_value), time_end - time_start)
+    else
+      if self.taggings_count_cache != real_value
+        self.taggings_count_cache = real_value
+        self.save
+      end
+    end
+    real_value
+  end
+
+  def check_if_taggings_count_needs_updating
```

Comment:
This isn't checking if the count needs updating, it's actually updating the count if it needs updating. `update_taggings_count` might suffice - which in turn begs the question what this method is doing that isn't or couldn't be covered already in `taggings_count`